tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -257,6 +257,60 @@ public function test_process_refund_on_uncaptured_payment() {
 		$this->assertEquals( 'uncaptured-payment', $result->get_error_code() );
 	}
 
+	public function test_process_refund_success_does_not_set_refund_failed_meta() {
+		$intent_id = 'pi_xxxxxxxxxxxxx';
+		$charge_id = 'ch_yyyyyyyyyyyyy';
+
+		$order = WC_Helper_Order::create_order();
+		$order->update_meta_data( '_intent_id', $intent_id );
+		$order->update_meta_data( '_charge_id', $charge_id );
+		$order->save();
+
+		$this->mock_api_client->expects( $this->once() )->method( 'refund_charge' )->will(
+			$this->returnValue(
+				[
+					'id'                       => 're_123456789',
+					'object'                   => 'refund',
+					'amount'                   => 19.99,
+					'balance_transaction'      => 'txn_987654321',
+					'charge'                   => 'ch_121212121212',
+					'created'                  => 1610123467,
+					'payment_intent'           => 'pi_1234567890',
+					'reason'                   => null,
+					'reciept_number'           => null,
+					'source_transfer_reversal' => null,
+					'status'                   => 'succeeded',
+					'transfer_reversal'        => null,
+					'currency'                 => 'usd',
+				]
+			)
+		);
+
+		$this->wcpay_gateway->process_refund( $order->get_id(), 19.99 );
+		$this->assertEquals( '', $order->get_meta( '_wcpay_refund_status', true ) );
+	}
+
+	public function test_process_refund_failure_sets_refund_failed_meta() {
+		$intent_id = 'pi_xxxxxxxxxxxxx';
+		$charge_id = 'ch_yyyyyyyyyyyyy';
+
+		$order = WC_Helper_Order::create_order();
+		$order->update_meta_data( '_intent_id', $intent_id );
+		$order->update_meta_data( '_charge_id', $charge_id );
+		$order->update_status( 'processing' );
+		$order->save();
+
+		$order_id = $order->get_id();
+
+		$this->mock_api_client
+			->expects( $this->once() )
+			->method( 'refund_charge' )
+			->willThrowException( new \Exception( 'Test message' ) );
+
+		$result = $this->wcpay_gateway->process_refund( $order_id, 19.99 );
+		$this->assertEquals( 'failed', $order->get_meta( '_wcpay_refund_status', true ) );
+	}
```

Comment:
For some reason this test is failing and I don't know why. I'd love some help with debugging it.

The reason it fails is that the order meta isn't updated; the `_wcpay_refund_status` remains empty, even though it's supposed to be set during the error handling in `process_refund`.

Reply:
This has been fixed in 32f8bb30d27cc3823dd43df7bd3ea761384316d9 🙌